compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -1751,6 +1762,26 @@ impl TyOrConstInferVar<'tcx> {
     }
 }
 
+/// Replace `{integer}` with `i32` and `{float}` with `f64`.
+/// Used only for diagnostics.
+struct InferenceLiteralEraser<'a, 'tcx> {
+    infcx: &'a InferCtxt<'a, 'tcx>,
```

Comment:
not that it matters too much, but why are u using `infcx` instead of only the `tcx` here